app/helpers/application_helper.rb

```diff
@@ -516,40 +516,32 @@ def category_badge(category, opts = nil)
     CategoryBadge.html_for(category, opts).html_safe
   end
 
-  def self.all_connectors
-    @all_connectors = Dir.glob("plugins/*/app/views/connectors/**/*.html.erb")
-  end
-
-  PLUGIN_OUTLET_TEMPLATE_CACHE = Concurrent::Map.new
+  SERVER_PLUGIN_OUTLET_PLUGINS_PREFIX = Rails.env.test? ? "spec/fixtures/plugins/" : "plugins/"
+  private_constant :SERVER_PLUGIN_OUTLET_PLUGINS_PREFIX
+
+  SERVER_PLUGIN_OUTLET_CONNECTOR_TEMPLATES =
+    Dir
+      .glob("#{SERVER_PLUGIN_OUTLET_PLUGINS_PREFIX}*/app/views/connectors/**/*.html.erb")
+      .reduce({}) do |connectors, template_path|
+        template_path =~ Regexp.new("/connectors/(.*)/.*\.html\.erb$")
+        outlet_name = Regexp.last_match(1)
+        connectors[outlet_name] ||= []
+        connectors[outlet_name] << template_path.sub(
+          "#{SERVER_PLUGIN_OUTLET_PLUGINS_PREFIX}",
+          "",
+        ).delete_suffix(".html.erb")
+        connectors
+      end
+  private_constant :SERVER_PLUGIN_OUTLET_CONNECTOR_TEMPLATES
```

Comment:
Files don't change once the application is loaded so I moved most of the logic out into a constant.